[Feat/schedule] serializers_fix

BH13KDR opened the pull request:
## ✅ PR 요약
- 작업 요약: shedule serializer를 요구사항에 맞게 더 수정하고, create 기능을 완성 후 test했습니다.

## 📄 상세 내용
- [ ] 주요 변경 사항 1 : participants를 따로 검증하던 serializer를 같이 검증하도록 변경했습니다.
- [ ] 주요 변경 사항 2 : merge이슈로 새로운 브랜치에서 PR을 하게 되었습니다.

chdan-hub committed:
✨ feat: 어드민 회원 상세 조회 serializer 코드 작성

chdan-hub committed:
🐛 fix: created_at에 format 수정

chdan-hub committed:
🐛 fix: 어드민 회원 정보 상세 조회용 시리얼라이저 수정

chdan-hub committed:
🐛 fix: 시리얼라이저 데이터 순서 변경

chdan-hub committed:
✨ feat: 유저 정보 상세 조회 view 코드 작성

chdan-hub committed:
✨ feat: 유저 정보 상세 조회 url 매핑

# Conflicts:
#	config/urls.py

# Conflicts:
#	config/urls.py

LuSiEDa committed:
✨ feat     : notification service 및 테스트 추

logger를 통해 로그를 확인할 수 있다.
기본세팅되어 있는 환경변수를 참조해서 redis를 세팅하고 채널,퍼블리쉬, 구독, 스트리밍 할 수 있게끔 함.

LuSiEDa committed:
🐛 fix: notification service test 오류 수정

LuSiEDa committed:
🐛 fix : isort 적용

chdan-hub committed:
🐛 fix: url 매핑 수정

chdan-hub committed:
🐛 fix: tags 이름 v1에서 Admin으로 변경

Commit:
url rebase

chdan-hub committed:
🐛 fix: exception_handler 적용

Commit:
💡 chore    : isort,black 포멧팅

LuSiEDa committed:
💡 chore : test에 피드백 사항 반영

Commit:
💡 chore    : logger 확인 코드 삭제

Commit:
# 🐛 fix      : branch실수로 새로 만들어진 PR용 브랜치입니다.# ✨ feat     : schedule-serializer 및 create에 한정한 간략한 view와 test가 들어있습니다.

Commit:
💡 chore: generrate_presigned_urls 복수 검증 추가

Meoyoug commented on apps/study_groups/serializers/schedule_serializers.py:
```diff
@@ -0,0 +1,84 @@
+        required=False,
+        write_only=True,
+    )
+
```
단순히 pk인 int값을 나열하는 ListField에 child 로 IntergerField를 두는 것 보다

PrimaryKeyRelatedField를 활용하시는게 좋아보입니다!

https://www.django-rest-framework.org/api-guide/relations/#primarykeyrelatedfield

Meoyoug commented on apps/study_groups/serializers/schedule_serializers.py:
```diff
@@ -0,0 +1,84 @@
+    def validate_objective(self, value: str | None) -> str | None:
+        if value is not None and len(value) > 500:
+            raise serializers.ValidationError("설명은 500자를 초과할 수 없습니다.")
+        return value
```
여기는 단순히 문자열 필드의 값이 존재하고 길이가 유효한지 판단하는 것 같은데
필드의 min_length, max_length, allow_blank 속성을 부여해서 검증할 수 있습니다

Commit:
💡 chore: isort , black 코드 포멧팅

Meoyoug commented on apps/study_groups/serializers/schedule_serializers.py:
```diff
@@ -0,0 +1,84 @@
+        if isinstance(value, datetime):
+            value_date = value.date()
+        else:
+            value_date = value
```
session_date 필드를 모델정의할 때 DateField로 지정하셨다면 무의미한 절인 것 같습니다.

Meoyoug commented on apps/study_groups/serializers/schedule_serializers.py:
```diff
@@ -0,0 +1,84 @@
+            value_date = value
+
+        if value_date < today:
+            raise serializers.ValidationError("날짜 설정이 잘못되었습니다.")
```
Validation 에러 메시지는 구체적으로 주세요!

session_data는 오늘보다 이전일 수 없습니다. 이런식으로요

Meoyoug commented on apps/study_groups/serializers/schedule_serializers.py:
```diff
@@ -0,0 +1,84 @@
+                h, m, s = map(int, end_time_str.split(":"))
+                end_time = dt_time(h, m, s)
+                if value >= end_time:
+                    raise serializers.ValidationError({"detail": "시간 설정이 잘못 되었습니다."})
```
여기도 어떤게 구체적으로 잘못되었는지?

start_time은 end_time과 같거나 이후일 수 없다.

이런식으로요

그리고 단순히 time으로 대소비교를 하면 1초만 빨라도 문제없이 되는데

최소한 시작과 종료가 10~30분 차이는 나는지 검증하는게 어떨까요?

Meoyoug commented on apps/study_groups/serializers/schedule_serializers.py:
```diff
@@ -0,0 +1,84 @@
+            user_id__in=participant_ids,
+        )
+        if members.count() != len(participant_ids):
+            raise serializers.ValidationError({"participants": "유효하지 않은 스터디 그룹 멤버가 포함되어 있습니다."})
```
```suggestion
        member_ids = GroupMember.objects.filter(study_group_id=study_group).values_list("id", flat=True)
        not_match_ids = [i for i in participant_ids if i not in member_ids]
        if not_match_ids:
            not_match_ids_str = ", ".join([str(i) for i in not_match_ids])
            raise serializers.ValidationError({"participants": f"유효하지 않은 스터디 그룹 멤버 id: {not_match_ids_str}"})
```

Meoyoug commented on apps/study_groups/services/schedule_services.py:
```diff
@@ -0,0 +1,22 @@
+class ScheduleService:
+    @staticmethod
+    def create_schedule(validated_data: dict[str, Any]) -> GroupSchedule:
+        print("create_schedule received:", validated_data)
```
PR 커밋시 print문은 제거해주세요

Meoyoug replied:
또는 Logger 활용해주세요

Meoyoug commented on apps/study_groups/services/schedule_services.py:
```diff
@@ -0,0 +1,22 @@
+            members = GroupMember.objects.filter(
+                study_group_id=schedule.study_group,
+                user_id__in=participant_ids,
+            )
```
이 쿼리가 굳이 필요했을 까 싶은데
create 시 member_id로 바로 넣어줄수있지않나용

Meoyoug commented on apps/study_groups/services/schedule_services.py:
```diff
@@ -0,0 +1,22 @@
+            bulk = [ScheduleParticipants(schedule=schedule, member=m) for m in members]
+            ScheduleParticipants.objects.bulk_create(bulk)
+
+        return schedule
```
이 메서드는 transaction처리가 필요해보이네요

Kang9805 committed:
✨ feat : 관리자용 지원서 목록/상세 조회 기능 구현

- AdminApplicationListView 신규 구현
- AdminApplicationDetailView 신규 구현
- Admin 전용 Serializer 구조 정리 및 상세 필드 추가
- application_urls.py 내부 구조 개선 및 admin_urls.py 연동
- 관리자 API 테스트 코드(test_admin_application_views.py) 추가
- 페이지네이션 임시 TODO 처리

Kang9805 committed:
💡 chore : admin_application_serializer 수정

AdminRecruitmentDetailSerializer의 타입 힌트 수정(Application → Recruitment)

chdan-hub committed:
🐛 fix: serializer role 및 status 유저 모델 property로 변경

chdan-hub committed:
🐛 fix: property에 따른 시리얼라이저 코드 변경

chdan-hub committed:
🐛 fix: config url 수정

Kang9805 committed:
♻️ refactor : admin application serializer의 lecture/tag 처리 방식 수정 및 mypy 오류 해결

- RecruitmentDetailSerializer에서 lectures, tags 조회 시 study_group → studylecture 연결 방식 적용
- recruitment_tags 기반 태그 조회 로직 정리
- SerializerMethodField 반환 타입을 Any로 변경하여 mypy 타입 충돌 해결
- admin application 목록/상세 조회 관련 prefetch 설정 보완

chdan-hub committed:
🐛 fix: status 조건 수정

Kang9805 committed:
- 페이지네이션 전용 클래스 추가(AdminApplicationPagination)
- 테스트 전부 통과하도록 검색 및 정렬 로직 보완
- 검색 기능 통합(search: title/nickname/email)
- status 필터링을 choices 기반으로 개선

Kang9805 committed:
♻️ refactor : 공통 QuerySet 분리 및 타입 힌트 추가로 코드 구조 개선

- admin application 조회 공통 쿼리셋(get_admin_application_queryset) 분리
- select_related / prefetch_related 중복 제거
- QuerySet[Application] 반환 타입 명시하여 mypy 오류 해결
- pagination 적용 시 paginated_qs로 시리얼라이저 변경
- 정렬(sort) 처리 로직 안정성 개선 (fallback 포함)

Commit:
# 🐛 fix      : PrimaryKeyRelatedField로 PK값을 받게 바꿨습니다. 문자열을 min,max_legth로 검증하게 했습니다, Validation 에러 메시지를 더 구체적으로 만들었습니다, 참가자 유효성 검사를 더 엄격하게 바꾸었습니다,service에 transaction처리로 양자성을 확보했습니다.

BH13KDR commented:
또 다른 분 커밋이 들어왔습니다.
